src/ProxyManager/ProxyGenerator/Util/PublicScopeSimulator.php

```diff
@@ -42,19 +43,21 @@ class PublicScopeSimulator
      * This is done by introspecting `debug_backtrace()` and then binding a closure to the scope
      * of the parent caller.
      *
-     * @param string            $operationType      operation to execute: one of 'get', 'set', 'isset' or 'unset'
-     * @param string            $nameParameter      name of the `name` parameter of the magic method
-     * @param string|null       $valueParameter     name of the `value` parameter of the magic method
-     * @param PropertyGenerator $valueHolder        name of the property containing the target object from which
-     *                                              to read the property. `$this` if none provided
-     * @param string|null       $returnPropertyName name of the property to which we want to assign the result of
-     *                                              the operation. Return directly if none provided
+     * @param \ReflectionMethod|null $originalMethod     the original method, if it exists
+     * @param string                 $operationType      operation to execute: one of 'get', 'set', 'isset' or 'unset'
+     * @param string                 $nameParameter      name of the `name` parameter of the magic method
+     * @param string|null            $valueParameter     name of the `value` parameter of the magic method
+     * @param PropertyGenerator      $valueHolder        name of the property containing the target object from which
+     *                                                   to read the property. `$this` if none provided
+     * @param string|null            $returnPropertyName name of the property to which we want to assign the result of
+     *                                                   the operation. Return directly if none provided
      *
      * @return string
      *
      * @throws \InvalidArgumentException
      */
     public static function getPublicAccessSimulationCode(
+        ?\ReflectionMethod $originalMethod,
```

Comment:
Also note: since we are passing down reflection details, we may as well get rid of a lot of runtime boilerplate, to be replaced with string constants.